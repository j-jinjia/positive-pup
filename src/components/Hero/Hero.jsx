import Button from "../Button/Button";
import "./Hero.scss";

const Hero = ({ header, subheaderText, buttonText, isSecondary, link }) => {
  return (
    <div className="hero">
      {header}
      <p className="hero__subheader">{subheaderText}</p>
      <a href="#mission-section">
        <Button buttonText={buttonText} isSecondary={isSecondary} link={link} />
      </a>
    </div>
  );
};

export default Hero;
