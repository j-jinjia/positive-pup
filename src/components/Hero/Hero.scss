@use "../../styles/variables" as *;
.hero {
  padding-bottom: 4rem;
  .hero-content {
    align-self: center;
    padding-top: 3rem;
    padding: 2rem;
    &__header {
      font-family: $font-header;
      font-size: $font-size-medium;
      font-weight: 500;
      position: relative;
      margin-bottom: 40px;

      &-highlight {
        display: inline;
        color: $colour-primary;
        font-family: inherit;
        font-size: inherit;
        position: relative;
        &::after {
          content: "";
          position: absolute;
          bottom: -0.5rem;
          left: 0;
          height: 0.6rem;
          width: 100%;
          background-image: url("../../assets/svgs/underline.svg");
          background-repeat: no-repeat;
          background-size: cover;
          z-index: -1;
        }
      }
    }
    &__subheader {
      font-size: $font-size-default;
      line-height: 150%;
      font-family: $font-general;
      color: hsla(259, 12%, 45%, 1);
      margin-bottom: 60px;
    }
  }
  .hero-image {
    background: none;
    padding: 0;
    margin: 0;
    &__image {
      display: none;
    }
  }
}
@media screen and (min-width: $tablet-breakpoint) {
  .hero {
    display: grid;
    grid-template-columns: 1fr;
    justify-items: center;
    .hero-content {
      text-align: center;
      padding-left: 5rem;
      padding-right: 5rem;
      &__header {
        font-size: $font-size-larger;
        margin-bottom: 26px;
      }
    }
    .hero-image {
      background: linear-gradient(
        to right,
        transparent 15%,
        rgba(108, 168, 162, 0.2) 15%,
        rgba(55, 175, 162, 0.2) 85%,
        transparent 85%
      );
      &__image {
        display: inline;
      }
    }
  }
}

@media screen and (min-width: $desktop-breakpoint) {
  .hero {
    display: grid;
    grid-template-columns: 1fr 1fr;
    .hero-image {
      display: flex;
      height: 450px;
      &__image {
        width: 100%;
        height: 100%;
        padding-bottom: 5rem;
      }
    }

    .hero-content {
      text-align: left;
    }

    .hero-certifications {
      grid-column-start: 2;
      display: flex;
    }
  }
}

@media screen and (min-width: $large-desktop-breakpoint) {
  .hero {
    display: grid;
    grid-template-columns: 1fr 1fr;
    .hero-content {
      &__subheader {
        font-size: $font-size-smallest;
      }
    }
    .hero-image {
      display: flex;
      padding-bottom: 6rem;
      height: 100%;
      &__image {
        width: 100%;
        height: 100%;
        padding-bottom: 5rem;
      }
    }

    .hero-content {
      text-align: left;
    }

    .hero-certifications {
      grid-column-start: 2;
      display: flex;
    }
  }
}
