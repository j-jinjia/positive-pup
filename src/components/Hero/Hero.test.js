import { customRender } from "../../utils/testUtils";
import Hero from "./Hero";

it("should match hero component snapshot", () => {
  const { container } = customRender(
    <Hero
      headingText={"About Us"}
      subheadingText={headerJSX}
      subheadingSecondLine={"My name is Melanie!"}
      isLeftAlign={leftAlign}
    />
  );

  expect(container).toMatchSnapshot();
});
