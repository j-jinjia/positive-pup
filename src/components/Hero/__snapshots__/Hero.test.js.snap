// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match hero component snapshot 1`] = `
<div>
  <div
    class="hero"
  >
    <div
      class="hero-content"
    >
      <p
        class="hero-content__subheader"
      />
      <button
        class="button button--primary"
      />
    </div>
    <div
      class="hero-image"
    >
      <img
        alt="Picture of a happy Staffie"
        class="hero-image__image"
      />
    </div>
    <div
      class="hero-certifications"
    >
      <div
        class="certifications"
      >
        <div
          class="certifications__badge-container"
        >
          <a
            href="https://companionanimal.network/"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="international companion animal network certification badge"
              class="certifications__badge certifications__badge--small"
              src="ICAN.png"
            />
          </a>
          <a
            href="https://www.petadvocacy.info/"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="Pet Industry Advocacy International badge"
              class="certifications__badge certifications__badge--small"
              src="PIAI.png"
            />
             
          </a>
          <a
            href="https://www.vsdogtrainingacademy.com/"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="Victoria Stillwell certification badge"
              class="certifications__badge certifications__badge--small"
              src="VSWA.png"
            />
          </a>
          <a
            href="https://ukdogcharter.org/"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="UK dog behaviour and training charter certification badge"
              class="certifications__badge certifications__badge--small"
              src="behave.png"
            />
          </a>
          <a
            href="https://www.petprofessionalguild.com/"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="the pet professional guild certification badge"
              class="certifications__badge certifications__badge--small"
              src="ppg2022.png"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
